components/sync15/README.md

```diff
@@ -5,4 +5,108 @@ data stores that want to sync against a Firefox Sync v1.5 sync server.
 It handles things like encrypting/decrypting records, obtaining and
 using storage node auth tokens, and so-on.
 
-It also needs a more complete README...
+There are 2 key concepts to understand here - the implementation itself, and
+a rust trait for a "syncable store" where component-specific logic lives - but
+before we dive into them, some preamble might help put things into context.
+
+## Introduction and History
+
+For many years Sync has worked exclusively against a "sync v1.5 server". This
+[is a REST API described here](https://mozilla-services.readthedocs.io/en/latest/storage/apis-1.5.html).
+The important part is that the API is conceptually quite simple - there are
+arbitrary "collections", indexed by a GUID, and lacking traditonal database
+concepts like joins. Because the payload is encrypted, there's very little
+scope for it to be much smarter. Thus it's reasonably easy to create a fairly
+generic abstraction over the API that can be easily reused.
+
+Back in the deep past, we found ourselves with 2 different components that
+needed to sync against a sync v1.5 server. The apps using these components
+didn't have schedulers or any UI for choosing what to sync - so these
+components just looked at the existing state of the engines on the server and
+synced if they were enabled.
+
+This was also pre-megazord - the idea was that apps could choose from a "menu"
+of components to include - so we didn't really want to bind these components
+together. Therefore, there was no concept of "sync all" - instead, each of the
+components had to be synced individually. So this component stated out as more
+of a "library" than a "component" which individual components could reuse - and
+each of these components was a "syncable store".
+
+Fast forward to Fenix and we needed a UI for managing all the engines supported
+there, and a single "sync now" experience etc - so we also have a sync_manager
```

Comment:
Is it also true that were were some bugs here when syncing just a single component, e.g. not respecting the user's data choices?

Reply:
I don't recall any - before the sync manager we never attempted to update the enabled state and only used what was already on the server. I guess there would have been an edge-case when the server was empty and a desktop was yet to upload the declined state it has persisted, but I'm not sure there's value trying to explain that in detail here.